notebooks/ch17/classification_sensitivity_specificity.md

```diff
@@ -0,0 +1,486 @@
+    return indicator_array
+```
+
+## Evaluating Classification Models
+
+Although we used the classification accuracy to evaluate our logistic model in previous sections, using the accuracy alone has some serious flaws that we explore in this section. To address these issues, we introduce a more useful metric to evaluate classifier performance: the **area under curve** metric.
+
+Suppose we have a dataset of 1000 emails that are labeled as spam or ham and our goal is to build a classifier that distinguishes future spam emails from ham emails. The data is contained in the `emails` DataFrame displayed below:
+
+
+```python
+emails
+```
+
+
+
+
+<div>
+<style scoped>
+    .dataframe tbody tr th:only-of-type {
+        vertical-align: middle;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>email</th>
+      <th>spam</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>0</th>
+      <td>\n Hi Folks,\n \n I've been trying to set a bu...</td>
+      <td>0</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>Hah.  I guess she doesn't want everyone to kno...</td>
+      <td>0</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>This article from NYTimes.com \n has been sent...</td>
+      <td>0</td>
+    </tr>
+    <tr>
+      <th>...</th>
+      <td>...</td>
+      <td>...</td>
+    </tr>
+    <tr>
+      <th>997</th>
+      <td>&lt;html&gt;\n &lt;head&gt;\n     &lt;meta http-equiv="Conten...</td>
+      <td>1</td>
+    </tr>
+    <tr>
+      <th>998</th>
+      <td>&lt;html&gt;\n &lt;head&gt;\n &lt;/head&gt;\n &lt;body&gt;\n \n &lt;cente...</td>
+      <td>1</td>
+    </tr>
+    <tr>
+      <th>999</th>
+      <td>\n &lt;html&gt;\n \n &lt;head&gt;\n &lt;meta http-equiv=3D"Co...</td>
+      <td>1</td>
+    </tr>
+  </tbody>
+</table>
+<p>1000 rows × 2 columns</p>
+</div>
+
+
+
+Each row contains the body of an email and a `spam` column, which is `0` if the email is ham or `1` if it is spam.
+
+Let's compare the performance of three different classifiers:
+* `ham_only`: predicts 'ham' for every email.
+* `spam_only`: predicts 'spam' for every email.
+* `words_list_model`: predicts 'ham' or 'spam' based on the presence of certain words in the body of an email. 
+
+We construct `words_list_model` using the following procedure: Suppose we have a list of words `words_list` that we believe are common in spam emails: "please", "click", "money", "business", and "remove". We transform each email into a feature vector by setting the vector's $i$th entry to 1 if the $i$th word in `words_list` is contained in the email body and 0 if it isn't. For example, using our five chosen words and the email body "please remove by tomorrow", the feature vector would be $[1, 0, 0, 0, 1]$. This procedure generates the `1000 X 5` feature matrix $X$.
```

Comment:
The first sentence has two colons in it; I think the first clause can be split off into its own sentence and ended with a period.